--- install.sh.orig	2025-10-10 10:07:35.835998523 +0400
+++ install.sh	2025-10-10 10:08:12.170468736 +0400
@@ -3,7 +3,7 @@
 
 # Argument parsing
 SHORT=r,d:,p:,s:,h
-LONG=remove,dest-dir:,prefix-dir:,sysconf-dir:,no-ectool,no-pre-uninstall,no-post-install,no-battery-sensors,no-sudo,no-pip-install,pipx,python-prefix-dir:,effective-installation-dir:,help
+LONG=remove,dest-dir:,prefix-dir:,sysconf-dir:,no-ectool,no-pre-uninstall,no-post-install,no-battery-sensors,no-sudo,no-pip-install,no-pip-build,pipx,python-prefix-dir:,effective-installation-dir:,help
 VALID_ARGS=$(getopt -a --options $SHORT --longoptions $LONG -- "$@")
 if [[ $? -ne 0 ]]; then
     exit 1;
@@ -22,6 +22,7 @@
 NO_BATTERY_SENSOR=false
 NO_SUDO=false
 NO_PIP_INSTALL=false
+NO_PIP_BUILD=false
 PIPX=false
 PYTHON_PREFIX_DIRECTORY_OVERRIDE=
 EFFECTIVE_INSTALLATION_DIRECTORY_OVERRIDE=
@@ -59,6 +60,10 @@
     '--no-sudo')
         NO_SUDO=true
         ;;
+    '--no-pip-build')
+        NO_PIP_BUILD=true
+        NO_PIP_INSTALL=true
+        ;;
     '--no-pip-install')
         NO_PIP_INSTALL=true
         ;;
@@ -96,7 +101,7 @@
 
 PYTHON_SCRIPT_INSTALLATION_PATH="$INSTALLATION_DIRECTORY/fw-fanctrl"
 
-if ! python3 -h 1>/dev/null 2>&1; then
+if ! python3 -h 1>/dev/null 2>&1 && [ "$NO_PIP_BUILD" = false ]; then
     echo "Missing package 'python3'!"
     exit 1
 fi
@@ -115,7 +120,7 @@
     fi
 fi
 
-if [ "$SHOULD_REMOVE" = false ]; then
+if [ "$SHOULD_REMOVE" = false ] && [ "$NO_PIP_BUILD" = false ]; then
     if ! python3 -m build -h 1>/dev/null 2>&1; then
         echo "Missing python package 'build'!"
         exit 1
@@ -232,7 +237,9 @@
     fi
     mkdir -p "$DEST_DIR$SYSCONF_DIR/fw-fanctrl"
 
-    build
+    if [ "$NO_PIP_BUILD" = false ]; then
+        build
+    fi
 
     if [ "$NO_PIP_INSTALL" = false ]; then
         echo "installing python package"
